lang/js/angular.md

```diff
@@ -0,0 +1,204 @@
+            $scope.title = 'Sessions';
+        };
+    });
+
+
+Controllers
+-----------
+
+
+### controllerAs with vm
+
+Supersedes the topics presented in the [controllerAs with vm](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#controlleras-with-vm) section.
+
+We don't require the use of using a variable like `vm` within a controller, properties and methods can be assigned to `$scope` when they need to. Just make sure to only assigne to `$scope` what needs to be accessible.
+
+
+### Bindable Members Up Top
+
+Supersedes some topics presented in the [Bindable Members](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#bindable-members-up-top) section.
+
+We don't want all the bindable members at the top if it calls a function as the called functions shouldn't be called before they are defined. Refer above for example on how to structure code in regards to placement of methods and properties.
+
+
+### Assigning Controllers
+
+Supersedes some topics presented in the [Assigning Controllers](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#assigning-controllers) section.
+
+If a controller is to be assigned within a view, it should be put into the template. This makes it easier and everything is contained within the template and module.
+
+    // config.js
+    "use strict";
+
+    function config($routeProvider) {
+        $routeProvider
+            .when("/games", {
+                templateUrl: "app/game/games.html"
+            });
+    }
+
+    angular.module("app").config(config);
+
+Template:
+
+    <!-- games.html -->
+    <div ng-controller="GamesController as games">
+    </div>
+
+
+Services and Factories
+----------------------
+
+Services and Factories are formatted the same way and only return the functions outside sources should have access to.
+
+    // factory.js
+    "use strict";
+
+    angular.module("app").factory("service", () => {
+        function function1 () {
+            /* */
+        }
+
+        function function2 () {
+            return function3();
+        }
+
+        function function3 () {
+            /* */
+        }
+
+        return {
+            function1,
+            function2
+        }
+    });
+
+
+Directives
+----------
+
+There is no need to format the directive by returning a variable. This is a waste of memory and lines as there is nothing being done with the variable except to return it.
+
+    // directive.js
+    "use strict";
+
+    function directive () {
+        return {
+            bindController: true,
+            controller: "ControllerController",
+            controllerAs: "vm",
+            restrict: "AE",
+            scope: {
+                // scope properties
+            },
+            templateUrl: "path/to/tempate.html"
+        }
+
+        angular.module("module").directive("directive", directive);
+    }
+
+
+Manual Annotating for Dependency Injection
+------------------------------------------
+
```

Comment:
A sentence or two should go here. Don't forget the margin-top on headings is set to 2em.